dataplane/standalone/translator.cc

```diff
@@ -83,25 +112,56 @@ sai_status_t Translator::remove(sai_object_type_t type, common_entry_t id) {
 sai_status_t Translator::set_attribute(sai_object_type_t type,
                                        sai_object_id_t id,
                                        const sai_attribute_t* attr) {
-  return sai_status_t();
-}
+  sai_object_id_t switch_id = this->attrMgr->get_switch_id(std::to_string(id));
+
+  sai_status_t status;
+  if (switch_id != SAI_NULL_OBJECT_ID) {
+    status = this->switches[switch_id]->set_child_attr(type, std::to_string(id),
+                                                       attr);
+  } else {
+    status = this->apis[std::to_string(id)]->set_attribute(attr);
+  }
+  if (status != SAI_STATUS_SUCCESS) {
+    return status;
+  }
 
-sai_status_t Translator::get_attribute(sai_object_type_t type,
-                                       sai_object_id_t id, uint32_t attr_count,
-                                       sai_attribute_t* attr_list) {
-  return sai_status_t();
+  this->attrMgr->set_attribute(std::to_string(id), attr);
+  return SAI_STATUS_SUCCESS;
 }
 
 sai_status_t Translator::set_attribute(sai_object_type_t type,
                                        common_entry_t id,
                                        const sai_attribute_t* attr) {
-  return sai_status_t();
+  std::string idStr = this->attrMgr->serialize_entry(type, id);
+  sai_object_id_t switch_id = this->attrMgr->entry_to_switch_id(type, id);
+
+  sai_status_t status;
+  if (switch_id != SAI_NULL_OBJECT_ID) {
+    status = this->switches[switch_id]->set_child_attr(type, idStr, attr);
+  } else {
+    status = this->apis[idStr]->set_attribute(attr);
+  }
+  if (status != SAI_STATUS_SUCCESS) {
+    return status;
+  }
+
+  this->attrMgr->set_attribute(idStr, attr);
```

Comment:
(no action required) Is it intended to call set_attribute twice?

Reply:
yes, one call is one the api (to perform some action if a attribute changes) then is the set is successful, commit the attribute to the new value